tools/llvm-pta-eval.cpp

```diff
@@ -0,0 +1,697 @@
+
+  if (n->pointsTo.empty()) {
+    puts(" -- no points-to");
+    return;
+  } else
+    putchar('\n');
+
+  for (const Pointer& ptr : n->pointsTo) {
+    printf("    -> ");
+    printName(ptr.target, false);
+    if (ptr.offset.isUnknown())
+      puts(" + UNKNOWN_OFFSET");
+    else
+      printf(" + %lu\n", *ptr.offset);
+  }
+  if (verbose) {
+    dumpPointerSubgraphData(n, type);
+  }
+}
+
+static void
+dumpPointerSubgraphdot(LLVMPointerAnalysis *pta, PTType type)
+{
+  std::set<PSNode *> nodes;
+  pta->getNodes(nodes);
+
+  printf("digraph \"Pointer State Subgraph\" {\n");
+
+  /* dump nodes */
+  for (PSNode *node : nodes) {
+    printf("\tNODE%p [label=\"", node);
+    printName(node, true);
+
+    if (node->getSize() || node->isHeap() || node->isZeroInitialized())
+      printf("\\n[size: %lu, heap: %u, zeroed: %u]",
+	     node->getSize(), node->isHeap(), node->isZeroInitialized());
+
+    if (verbose && node->getOperandsNum() > 0) {
+      printf("\\n--- operands ---\\n");
+      for (PSNode *op : node->getOperands()) {
+	printName(op, true);
+	printf("\\n");
+      }
+      printf("------\\n");
+    }
+
+    for (const Pointer& ptr : node->pointsTo) {
+      printf("\\n    -> ");
+      printName(ptr.target, true);
+      printf(" + ");
+      if (ptr.offset.isUnknown())
+	printf("UNKNOWN_OFFSET");
+      else
+	printf("%lu", *ptr.offset);
+    }
+
+    if (verbose)
+      dumpPointerSubgraphData(node, type, true /* dot */);
+
+    printf("\"");
+    if (node->getType() != STORE) {
+      printf(", shape=box");
+      if (node->pointsTo.size() == 0
+	  && (node->getType() == LOAD ||
+	      node->getType() == GEP))
+	printf(", style=filled, fillcolor=red");
+    } else {
+      printf(", shape=cds");
+    }
+
+    printf("]\n");
+  }
+
+  /* dump edges */
+  for (PSNode *node : nodes) {
+    for (PSNode *succ : node->getSuccessors())
+      printf("\tNODE%p -> NODE%p [penwidth=2]\n", node, succ);
+  }
+
+  printf("}\n");
+}
+
+static void
+dumpPointerSubgraph(LLVMPointerAnalysis *pta, PTType type, bool todot)
+{
+  assert(pta);
+
+  if (todot)
+    dumpPointerSubgraphdot(pta, type);
+  else {
+    std::set<PSNode *> nodes;
+    pta->getNodes(nodes);
+
+    for (PSNode *node : nodes) {
+      dumpPSNode(node, type);
+    }
+  }
+}
+
+typedef int AliasResult;
+#define NoAlias 1
+#define MayAlias 2
+#define MustAlias 3
+#define PartialAlias 4
+
+static int compare_pointer(const Pointer& ptr1, 
+			  const Pointer& ptr2)
+{
+  //  if (ptr1.target == &NullMemoryObject) return MayAlias;
+  //  if (ptr2.target == &NullMemoryObject) return MayAlias;
+  printf("target1=");
+  printName(ptr1.target, 0);
+  printf("\n");
+  printf("target2=");
+  printName(ptr2.target, 0);
+  printf("\n");
+  if (ptr1.target->isUnknownMemory()) return MayAlias;
+  if (ptr2.target->isUnknownMemory()) return MayAlias;
+  if (ptr1.offset == UNKNOWN_OFFSET) return MayAlias;
+  if (ptr2.offset == UNKNOWN_OFFSET) return MayAlias;
```

Comment:
Could you, please, keep the project's coding style and put the return statement on the next line?